docs/modules/architecture/pages/compact-binary-specification.adoc

```diff
@@ -0,0 +1,719 @@
+
+== Data Types
+
+The table below shows the data types that are supported by compact serialization.
+
+Endianness for primitives is configurable and by default, they are written in BIG_ENDIAN. This configuration is applied to user types and also all the primitives used in the format. These include the size of the arrays, number of items, length of the data, schema-id, offsets, etc. The endianness is inherited from the existing endianness configuration, a new endianness config isn't introduced for compact.
+
+[cols="1,1,1,1,1,1,1,1,3,1"]
+|===
+|Type |Java |C++ |C# |Python |Node.js |Go |SQL |Description| Fixed Size
+|boolean |boolean |bool |bool |bool |boolean |bool |BOOLEAN |true or false represented by 1 bit as either 1 or 0. Up to 8 booleans packed into a single byte |yes
+|int8
+|byte
+|int8_t
+|sbyte
+|int
+|number
+|int8
+|TINYINT
+|8 bit two's complement signed integer
+|yes
+|int16
+|short
+|int16_t
+|short
+|int
+|number
+|int16
+|SMALLINT
+|16-bit two's-complement signed integer
+|yes
+|int32
+|int
+|int32_t
+|int
+|int
+|number
+|int32
+|INTEGER
+|32-bit two's-complement signed integer
+|yes
+|int64
+|long
+|int64_t
+|long
+|int
+|Long
+|int64
+|BIGINT
+|64-bit two's-complement signed integer
+|yes
+|float32
+|float
+|float
+|float
+|float
+|number
+|float32
+|REAL
+|32-bit IEEE 754 floating-point number
+|yes
+|float64
+|double
+|double
+|double
+|float
+|number
+|float64
+|DOUBLE
+|64-bit IEEE 754 floating-point number
+|yes
+|string
+|String
+|optional<std::string>
+|string?
+|typing.Optional[str]
+|string \| null
+|*string
+|STRING
+|null or number of bytes in the string(iint32) + UTF-8 string https://tools.ietf.org/html/rfc3629
+|no
+|decimal
+|BigDecimal
+|optional<hazelcast:client:decimal>
+|HBigDecimal?
+|typing.Optional[decimal.Decimal]
+|BigDecimal \| null
+|*types.Decimal
+|DECIMAL
+|null or +
+Arbitrary precision and scale floating-point number: +
+represented as unscaledValue x 10 ^ -scale +
+unscaledValue: Array of int8 (byte array containing the two's-complement binary +
+representation in big-endian byte-order: the most significant byte is in the zeroth element.) +
+scale : single int32 for scale
+|no
+|time
+|LocalTime
+|optional<hazelcast:client:local_time
+|HLocalTime?
+|typing.Optional[datetime.time]
+|LocalTime \| null
+|*types.LocalTime
+|TIME
+|null or +
+HH-MI-SS-NN +
+int8: hour +
+int8: minute +
+int8: seconds +
+int32: nanoseconds +
+|no(since it is nullable)
+|date
+|LocalDate
+|optional<hazelcast:client:local_date>
+|HLocalDate?
+|typing.Optional[datetime.date]
+|LocalDate \| null
+|*types.LocalDate
+|DATE
+|null or +
+YYYY-MM-DD from -999999999-01-1 to 999999999-12-31 +
+int32: year +
+ int8: month +
+int8: dayOfMonth
+|no(since it is nullable)
+|timestamp
+|LocalDateTime
+|optional<hazelcast:client:local_date_time>
+|HLocalDateTime?
+|typing.Optional[datetime.datetime]
```

Comment:
for python, can we get rid of `typing.` prefix, just like the table we have in the https://docs.hazelcast.com/hazelcast/latest/serialization/compact-serialization#supported-types?